open_discussions_api/users/client.py

```diff
@@ -0,0 +1,51 @@
+"""Users API"""
+import json
+
+from open_discussions_api.base import BaseApi
+
+SUPPORTED_USER_ATTRIBUTES = (
+    'name',
+    'image',
+    'image_small',
+    'image_medium',
+)
+
+
+class UsersApi(BaseApi):
+    """Users API"""
+
+    def list(self):
+        """Returns a list of users"""
+        return self.session.get(self.get_url("/users/"))
+
+    def get(self, username):
+        """Gets a specific user"""
+        return self.session.get(self.get_url("/users/{}").format(username))
+
+    def create(self, **profile):
+        """Creates a new user"""
+        if not profile:
+            raise AttributeError("No fields provided to create")
+
+        for key in profile:
+            if key not in SUPPORTED_USER_ATTRIBUTES:
+                raise AttributeError("Argument {} is not supported".format(key))
+
+        return self.session.post(
+            self.get_url("/users/"),
+            data=json.dumps(dict(profile=profile or {}))
+        )
+
+    def update(self, username, **profile):
+        """Gets a specific user"""
+        if not profile:
+            raise AttributeError("No fields provided to update")
+
+        for key in profile:
+            if key not in SUPPORTED_USER_ATTRIBUTES:
+                raise AttributeError("Argument {} is not supported".format(key))
+
+        return self.session.patch(
+            self.get_url("/users/{}/".format(username)),
+            data=json.dumps(dict(profile=profile))
+        )
```

Comment:
That behavior is correct. The usernames get generated on the `open-discussions` side and are a random id. So if you're requesting to patch a username that doesn't exist than it's just an invalid request.

Reply:
Sorry, wasn't clear - my OD has a `username` but lacks a `Profile`. It might be just because that user was created a while ago - not sure if a `Profile` is automatically added now during user creation, or if that only happens for users created through the API.

Reply:
Oh ok, hmm, yeah, generally I'd say that we wouldn't create a user without a profile going forward, but there's nothing that really enforces that so it's probably safer. I'll fix.

Reply:
Filed as mitodl/open-discussions#122